cfme/common/host_views.py

```diff
@@ -0,0 +1,365 @@
+
+    def get_item(self, by_name=None, surf_pages=False):
+        """Obtains one item matched to by_name. Raises exception if no items or several items were
+           found.
+
+        Args:
+            by_name (str): only item which match to by_name will be returned
+            surf_pages (bool): current page items if False, all items otherwise
+
+        Returns: matched item (QuadIcon/etc.)
+        """
+        items = self.get_items(by_name=by_name, surf_pages=surf_pages)
+        if not items:
+            raise ItemNotFound("Item {name} isn't found on this page".format(name=by_name))
+        elif len(items) > 1:
+            raise ManyItemsFound("Several items with {name} were found".format(name=by_name))
+        return items[0]
+
+
+class HostSideBar(View):
+    """Represents left side bar. It usually contains navigation, filters, etc."""
+    pass
+
+
+class HostsView(ComputeInfrastructureHostsView):
+    toolbar = View.nested(HostsToolbar)
+    sidebar = View.nested(HostSideBar)
+    items = View.nested(HostItems)
+    paginator = View.nested(PaginationPane)
+
+    @property
+    def is_displayed(self):
+        return self.in_compute_infrastructure_hosts and self.title.text == "Hosts"
+
+
+class HostFormView(ComputeInfrastructureHostsView):
+    # Info/Settings
+    title = Text(".//div[@id='main-content']//h1")
+    name = Input(name="name")
+    hostname = Input(name="hostname")
+    custom_ident = Input(name="custom_1")
+    ipmi_address = Input(name="ipmi_address")
+    mac_address = Input(name="mac_address")
+
+    @View.nested
+    class endpoints(View):  # noqa
+        @View.nested
+        class default(Tab):  # noqa
+            username = Input(name="default_userid")
+            password = Input(name="default_password")
+            confirm_password = Input(name="default_verify")
+            validate_button = Button("Validate")
+
+        @View.nested
+        class remote_login(Tab):  # noqa
+            TAB_NAME = "Remote Login"
+            username = Input(name="remote_userid")
+            password = Input(name="remote_password")
+            confirm_password = Input(name="remote_verify")
+            validate_button = Button("Validate")
+
+        @View.nested
+        class web_services(Tab):  # noqa
+            TAB_NAME = "Web Services"
+            username = Input(name="ws_userid")
+            password = Input(name="ws_password")
+            confirm_password = Input(name="ws_verify")
+            validate_button = Button("Validate")
+
+        @View.nested
+        class ipmi(Tab):  # noqa
+            TAB_NAME = "IPMI"
+            username = Input(name="ipmi_userid")
+            password = Input(name="ipmi_password")
+            confirm_password = Input(name="ipmi_verify")
+            validate_button = Button("Validate")
+
+    cancel_button = Button("Cancel")
+
+
+class HostAddView(HostFormView):
+    host_platform = BootstrapSelect("user_assigned_os")
+    add_button = Button("Add")
```

Comment:
We don't have naming conventions yet  but add/cancel/save/etc are used everywhere.  It makes sense to follow one approach everywhere.